tests/e2e.rs

```diff
@@ -4,33 +4,104 @@ mod harness;
 
 use bitcoin::Amount;
 use bitcoin_harness::{self, Bitcoind};
-use harness::{create, update};
-use thor::{punish, update::ChannelUpdate};
+use futures::{
+    channel::mpsc::{Receiver, Sender},
+    SinkExt, StreamExt,
+};
+use harness::{create, update, Wallet};
+use thor::{punish, update::ChannelUpdate, Channel, ReceiveMessage, SendMessage};
+
+struct Network {
```

Comment:
```suggestion
struct Transport {
```
nitpick because in our context, `Network` can refer to the Bitcoin network.